File: 조준장/level3/네트워크.py
def solution(n, computers):
    answer = 0
    visited = []
    queue = []
    
    for i in range(n):
        if i not in visited:
            queue.append(i)
            answer += 1
            
            while queue:
                target = queue.pop()
                
                for j in range(n):
                    if computers[target][j] == 1 and j not in visited:
                        queue.append(j)
                        visited.append(j)
                        
    return answer

File: 조준장/level3/네트워크.kt
class Solution {
    fun solution(n: Int, computers: Array<IntArray>): Int {
        var answer = 0
        val queue = mutableListOf<Int>()
        val visited = mutableListOf<Int>()
        
        (0 until n).forEach{
            if(!visited.contains(it)){
                queue.add(it)
                answer ++ 
                while (!queue.isEmpty()){

                    var target = queue.removeAt(0)
                    
                    for (i in 0 until n){
                        if (computers[target][i] == 1 && !visited.contains(i)){
                            visited.add(i)
                            queue.add(i)
                        }
                    }
                }
            }
        }
        
        return answer
    }
}
